cmd/argo-watcher/server/router.go

```diff
@@ -265,3 +287,75 @@ func (env *Env) healthz(c *gin.Context) {
 func (env *Env) getConfig(c *gin.Context) {
 	c.JSON(http.StatusOK, env.config)
 }
+
+// SetDeployLock godoc
+// @Summary Set deploy lock
+// @Description Set deploy lock
+// @Tags frontend
+// @Success 200 {string} string
+// @Router /api/v1/deploy-lock [post].
+func (env *Env) SetDeployLock(c *gin.Context) {
+	if err := env.validateKeycloakToken(c); err != nil {
+		log.Error().Msgf("couldn't release deploy lock, got the following error: %s", err)
+		c.JSON(http.StatusUnauthorized, models.TaskStatus{
+			Status: unathorizedMessage,
+		})
+		return
+	}
+
+	env.deployLockSet = true
+
+	log.Debug().Msg("deploy lock is set")
+	c.JSON(http.StatusOK, "deploy lock is set")
+}
+
+// ReleaseDeployLock godoc
+// @Summary Release deploy lock
+// @Description Release deploy lock
+// @Tags frontend
+// @Success 200 {string} string
+// @Router /api/v1/deploy-lock [delete].
+func (env *Env) ReleaseDeployLock(c *gin.Context) {
+	if err := env.validateKeycloakToken(c); err != nil {
+		log.Error().Msgf("couldn't release deploy lock, got the following error: %s", err)
+		c.JSON(http.StatusUnauthorized, models.TaskStatus{
+			Status: unathorizedMessage,
+		})
+		return
+	}
+
+	env.deployLockSet = false
```

Comment:
Do we want to abstract this variable into a separate module / object? `DeployLocker` may be. 

In case we want to expand the functionality of it